libnmstate/state.py

```diff
@@ -258,11 +277,22 @@ def merge_route_config(self, other_state):
                     self_iface_route_sets[route[Route.NEXT_HOP_INTERFACE]].add(
                         RouteEntry(route))
 
+        # Check whether user desire routes to down interface/ip.
+        _validate_routes(self_iface_route_sets,
+                         iface_enable_states,
+                         ipv4_enable_states,
+                         ipv6_enable_states)
+
         for route in other_state._config_routes:
             # The other_state(running state) does not have any absent
             # route entry and all have NEXT_HOP_INTERFACE
-            iface_name = route[Route.NEXT_HOP_INTERFACE]
-            self_iface_route_sets[iface_name].add(RouteEntry(route))
+            if _route_is_valid(route,
+                               iface_enable_states,
+                               ipv4_enable_states,
+                               ipv6_enable_states):
+                # Don't copy route from down interface or disabled IP
```

Comment:
This comment should be added to the `_route_is_valid()` docstring. For me it is confusing at this postion.